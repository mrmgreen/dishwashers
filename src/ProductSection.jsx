import React from 'react';

function ProductSection(props) {
  const { title, price: { now }, image } = props.product;
  let priceToPay;

  if (typeof now === "object") {
    priceToPay = now.to;
  }

  return ( 
    <div className="productSection">
      <div className="product_section_top">
        <img src={image} alt="dishwasher"/>
      </div>
      <div className="product_section_bottom">
        <div className='product_title'>{title}</div>
        <div className='price'>Â£{ priceToPay || now}</div>
      </div>
    </div>
  )
}

export default ProductSection;