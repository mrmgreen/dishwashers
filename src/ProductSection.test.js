import React from 'react';
import { shallow } from 'enzyme';
import ProductSection from './ProductSection';

const product = {
  productId: 1391191,
  title:"Indesit DIF 04B1 Ecotime Fully Integrated Dishwasher, White",
  price:{  
    was:"",
    then1:"",
    then2:"",
    now:"220.00",
    uom:"",
    currency:"GBP"
  },
  image: "//johnlewis.scene7.com/is/image/JohnLewis/233326789?",
}

it('renders without crashing', () => {
  const component = shallow(<ProductSection />)
  expect(component).toHaveLength(1);
});

it('renders a product info', () => {
  const component = shallow(<ProductSection product={product} />);
  expect(component).toMatchSnapshot();
});

it('renders a price from now.to field', () => {
  const productToPrice = { ...product, price:{ now: { from:'20.00', to: '19.00' }}}
  const component = shallow(<ProductSection product={productToPrice} />);
  expect(component).toMatchSnapshot();
});