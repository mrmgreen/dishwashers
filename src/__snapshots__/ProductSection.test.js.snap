// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a price from now.to field 1`] = `
<div
  className="product__section"
>
  <Link
    to="/productpage/1391191"
  >
    <div
      className="product__section--top"
    >
      <img
        alt="dishwasher"
        src="//johnlewis.scene7.com/is/image/JohnLewis/233326789?"
      />
    </div>
    <div
      className="product__section--bottom"
    >
      <div
        className="product__title"
      >
        Indesit DIF 04B1 Ecotime Fully Integrated Dishwasher, White
      </div>
      <div
        className="price"
      >
        £
        19.00
      </div>
    </div>
  </Link>
</div>
`;

exports[`renders a product info 1`] = `
<div
  className="product__section"
>
  <Link
    to="/productpage/1391191"
  >
    <div
      className="product__section--top"
    >
      <img
        alt="dishwasher"
        src="//johnlewis.scene7.com/is/image/JohnLewis/233326789?"
      />
    </div>
    <div
      className="product__section--bottom"
    >
      <div
        className="product__title"
      >
        Indesit DIF 04B1 Ecotime Fully Integrated Dishwasher, White
      </div>
      <div
        className="price"
      >
        £
        220.00
      </div>
    </div>
  </Link>
</div>
`;
